prototype/local-factory/state.go

```
@@ -0,0 +1,210 @@
+	}
+	body = append(body, '\n')
+	return atomicWrite(filepath.Join(directory, "work.json"), body, 0o644)
+}
+
+func atomicWrite(path string, body []byte, mode os.FileMode) error {
+	temporary, err := os.CreateTemp(filepath.Dir(path), ".tmp-*")
+	if err != nil {
+		return err
+	}
+	temporaryPath := temporary.Name()
+	defer os.Remove(temporaryPath)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Remove` is not checked

(errcheck)